tracker/event_tracker_store_test.go

```diff
@@ -1,35 +1,177 @@
+
+	assert.NoError(t, err)
+	assert.Len(t, logs, 3)
+	assert.Equal(t, common.EncodeUint64ToBytes(2), key)
+
+	err = entry.saveNextToProcessIndx(1) // next time should start from the second one
 	require.NoError(t, err)
 
-	path := filepath.Join(dir, "test.db")
-	store, err := NewEventTrackerStore(path, 2, nil, hclog.Default())
+	logs, key, err = entry.getFinalizedLogs(14)
+
+	assert.NoError(t, err)
+	assert.Len(t, logs, 4)
+	assert.Equal(t, common.EncodeUint64ToBytes(4), key)
+}
+
+func Test_Entry_saveNextToProcessIndx(t *testing.T) {
+	const someFilterHash = "test"
+
+	tstore, fn := createSetupDB(nil, 2)(t)
+	defer fn()
+
+	entry, err := tstore.(*EventTrackerStore).getImplEntry(someFilterHash)
 	require.NoError(t, err)
 
-	closeFn := func() {
-		require.NoError(t, os.RemoveAll(dir))
+	entry.StoreLogs([]*ethgo.Log{
+		{BlockNumber: 1}, {BlockNumber: 5}, {BlockNumber: 8}, {BlockNumber: 11}, {BlockNumber: 12}, {BlockNumber: 15},
+	})
+
+	for i := 0; i < 10; i++ {
+		require.NoError(t, entry.saveNextToProcessIndx(uint64(i)))
 	}
+}
+
+func Test_EventTrackerStore_SetNotLastBlock(t *testing.T) {
+	t.Parallel()
+
+	subs := &mockEventSubscriber{}
 
-	return store, closeFn
+	tstore, fn := createSetupDB(subs, 2)(t)
+	defer fn()
+
+	assert.NoError(t, tstore.(*EventTrackerStore).Set("dummy", "dummy")) //nolint
+	assert.Len(t, subs.logs, 0)
 }
 
-func TestBoltDBStore(t *testing.T) {
+func Test_EventTrackerStore_onNewBlockBad(t *testing.T) {
 	t.Parallel()
 
-	store.TestStore(t, setupDB)
+	tstore, fn := createSetupDB(nil, 0)(t)
+	defer fn()
+
+	assert.Error(t, tstore.(*EventTrackerStore).onNewBlock("dummy", "dummy"))                          //nolint
+	assert.Error(t, tstore.(*EventTrackerStore).onNewBlock("dummy", hex.EncodeToString([]byte{0, 1}))) //nolint
+}
+
+func Test_EventTrackerStore_OnNewBlockNothingToProcess(t *testing.T) {
+	t.Parallel()
+
+	subs := &mockEventSubscriber{}
+
+	tstore, fn := createSetupDB(subs, 10)(t)
+	defer fn()
+
+	block := ethgo.Block{Number: 8}
+
+	bytes, err := block.MarshalJSON()
+	require.NoError(t, err)
+
+	value := hex.EncodeToString(bytes)
+
+	// block less than finality depth
+	assert.NoError(t, tstore.(*EventTrackerStore).onNewBlock("dummy", value)) //nolint
+	assert.Len(t, subs.logs, 0)
+
+	block = ethgo.Block{Number: 12}
+
+	bytes, err = block.MarshalJSON()
+	require.NoError(t, err)
+
+	value = hex.EncodeToString(bytes)
+
+	// no logs
+	assert.NoError(t, tstore.(*EventTrackerStore).onNewBlock("dummy", value)) //nolint
+	assert.Len(t, subs.logs, 0)
+}
+
+func Test_EventTrackerStore_SetLastBlockSubscriberNotified(t *testing.T) {
+	t.Parallel()
+
+	const hash = "dummy_hash"
+
+	subs := &mockEventSubscriber{}
+
+	tstore, fn := createSetupDB(subs, 2)(t)
+	defer fn()
+
+	entry, err := tstore.GetEntry(hash)
+	require.NoError(t, err)
+
+	require.NoError(t, entry.StoreLogs([]*ethgo.Log{
+		{BlockNumber: 1}, {BlockNumber: 2}, {BlockNumber: 3},
+	}))
+
+	for i := 0; i < 3; i++ {
+		block := ethgo.Block{Number: uint64(i + 2)}
```

Comment:
Could you add a bit more description on how this is testing the finality?

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1228/commits/58898271b21757ec25be593a5fcc7a983c032392